CameraBrowser/CameraBrowserSample/ViewController.cs

```diff
@@ -0,0 +1,238 @@
+#nullable enable
+
+namespace CameraBrowserSample
+{
+	public partial class ViewController : NSViewController, IICDeviceBrowserDelegate, IICCameraDeviceDelegate, IICCameraDeviceDownloadDelegate, IICDeviceDelegate {
+		public ICDeviceBrowser DeviceBrowser { get; set; } = new ICDeviceBrowser();
+		public NSMutableArray Cameras { [Export("Cameras")] get; [Export("setCameras:")] set; } = new NSMutableArray();
+
+		static ViewController() //init
+		{
+			var imageTransformer = new CGImageRefToNSImageTransformer();
+			NSValueTransformer.SetValueTransformer(imageTransformer, "CGImageRefToNSImage");
+		}
+
+		public ViewController (IntPtr handle) : base(handle)
+		{
+		}
+
+		public override void ViewWillAppear()
+		{
+			base.ViewWillAppear();
+			View.Window.Title = "Camera Browser Sample for Xamarin.Mac ";
+		}
+
+		public void HandleProcessICLaunchParams(NSNotification note) => Console.WriteLine($"NSNotification: {note}");
+
+		public override void ViewDidLoad () //application did finish launching
+		{
+			base.ViewDidLoad ();
+
+			// Do any additional setup after loading the view.
+
+			CamerasController.SelectsInsertedObjects = false;
+
+			NSNotificationCenter.DefaultCenter.AddObserver(this, new Selector("processICLaunchParams:"), (NSString)"ICLaunchParamsNotification", null);
+
+			CamerasController.SelectsInsertedObjects = false;
+
+			MediaFilesController.AddObserver(this, (NSString)"selectedObjects", (NSKeyValueObservingOptions)0, IntPtr.Zero);
+
+			DeviceBrowser.Delegate = this;
+			DeviceBrowser.BrowsedDeviceTypeMask = ICBrowsedDeviceType.Local | ICBrowsedDeviceType.Remote | ICBrowsedDeviceType.Camera;
+			DeviceBrowser.Start();
+		}
+
+		[Export("observeValueForKeyPath:ofObject:change:context:")]
+		private void ObserveValueForKeyPath (NSString keyPath, NSArrayController ofObject, NSDictionary change, IntPtr context)
+		{
+			if (keyPath == "selectedObjects" && ofObject == MediaFilesController)
+			{
+				WillChangeValue("canDelete");
+				WillChangeValue("canDownload");
+				DidChangeValue("canDelete");
+				DidChangeValue("canDownload");
+
+			}
+		}
+
+		[Export("canDelete")]
+		public bool CanDelete ()
+		{
+			bool can = false;
+			InvokeOnMainThread(() => {
+				var selectedFiles = MediaFilesController.SelectedObjects;
+				foreach (ICCameraFile f in selectedFiles)
+				{
+					if (!f.Locked)
+					{
+						can = true;
+						break;
+					}
+				}
+			});
+			return can;
+		}
+
+		[Export("canDownload")]
+		public bool CanDownload()
+		{
+			var res = false;
+			InvokeOnMainThread(() => {
+				res = MediaFilesController.SelectedObjects is not null ? MediaFilesController.SelectedObjects.Length > 0 : false;
+			} );
+			return res;
+		}
+
+		[Export("selectedCamera")]
+		public ICCameraDevice? SelectedCamera
+		{
+			get
+			{
+				var selectedObjects = CamerasController.SelectedObjects;
+				if (selectedObjects is null || selectedObjects.Length == 0)
+					return null;
+				return selectedObjects[0] as ICCameraDevice;
```

Comment:
This can be simplified with LINQ too:

```suggestion
				return CamerasController.SelectedObjects?.FirstOrDefault () as ICCameraDevice;
```